This is the jupyter notebook which will hold our neural networks and be used to create graphs and stuff

In [1]:
import numpy as np
from load import *
sets = load_data()
labels = ["X_test",
"y_test",
"person_train_valid",
"X_train_valid",
"y_train_valid",
"person_test"]
print('generating data using all test and all train')
for i, k in enumerate(sets):
    print('{}: {} '.format(labels[i], k.shape))
print('generating data using subj 1 test and subj 1 train')
printy = load_data_subject_1_train_and_test()
for i, k in enumerate(printy):
    print('{}: {} '.format(labels[i], k.shape))
print('generating data using subj 1 test and full train')
thingy = load_data_subject_1_test_and_full_train()
for i, k in enumerate(thingy):
    print('{}: {} '.format(labels[i], k.shape))

generating data using all test and all train
X_test: (443, 22, 1000) 
y_test: (443,) 
person_train_valid: (2115, 1) 
X_train_valid: (2115, 22, 1000) 
y_train_valid: (2115,) 
person_test: (443, 1) 
generating data using subj 1 test and subj 1 train
X_test: (50, 22, 1000) 
y_test: (50,) 
person_train_valid: (2115, 1) 
X_train_valid: (237, 22, 1000) 
y_train_valid: (237,) 
person_test: (443, 1) 
generating data using subj 1 test and full train
X_test: (50, 22, 1000) 
y_test: (50,) 
person_train_valid: (2115, 1) 
X_train_valid: (2115, 22, 1000) 
y_train_valid: (2115,) 
person_test: (443, 1) 


There are 2115 trials and 1000 timestamps. There are 22 electrodes. There are 443 test trials. There are persons 0-8 (inclusive) and person_train_valid[trial] gives the person that was used in the specific trial. 

The following cell takes the y data and subtracts 769 so the classes are 0, 1, 2, 3. DO NOT RUN THIS MORE THAN ONCE PER KERNEL RESTART.

train a NN for the first subject

In [70]:
from models import *
model = make_cnn_model(s1_x_valid, s1_y_valid, s1_x_test, s1_y_test)

ModuleNotFoundError: No module named 'tensorflow'